ssg/build_remediations.py

```diff
@@ -262,6 +269,46 @@ class BashRemediation(Remediation):
     def __init__(self, file_path):
         super(BashRemediation, self).__init__(file_path, "bash")
 
+    def parse_from_file_with_jinja(self, env_yaml):
+        self.local_env_yaml.update(env_yaml)
+        result = super(BashRemediation, self).parse_from_file_with_jinja(self.local_env_yaml)
+
+        rule_platforms = set()
+        if self.associated_rule:
+            # There can be repeated inherited platforms and rule platforms
+            rule_platforms.update(self.associated_rule.inherited_platforms)
+            rule_platforms.add(self.associated_rule.platform)
+
+        platform_conditionals = []
+        for platform in rule_platforms:
+            if platform == "machine":
+                # Based on check installed_env_is_a_container
+                platform_conditionals.append('[ ! -f /.dockerenv -a ! -f /run/.containerenv ]')
+            elif platform is not None:
+                # Assume any other platform is a Package CPE
+
+                # Some package names are different from the platform names
+                if platform in self.local_env_yaml["platform_package_overrides"]:
+                    platform = self.local_env_yaml["platform_package_overrides"].get(platform)
+
+                # Adjust package check command according to the pkg_manager
+                pkg_manager = self.local_env_yaml["pkg_manager"]
+                pkg_check_command = PKG_MANAGER_TO_PACKAGE_CHECK_COMMAND[pkg_manager]
+                platform_conditionals.append(pkg_check_command.format(platform))
+
+        if platform_conditionals:
+            platform_fix_text = "# Remediation is applicable only in certain platforms\n"
+
+            cond = platform_conditionals.pop(0)
+            platform_fix_text += "if {}".format(cond)
+            for cond in platform_conditionals:
+                platform_fix_text += " && {}".format(cond)
+            platform_fix_text += '; then\n{}\nelse\necho "Remediation is not applicable, nothing was done"\nfi'.format(result.contents)
```

Comment:
1. I would use `textwrap. indent` to indent the innner original body of the remediation if it doesn't break our dark remediation generation magic.
2. What do you think of redirecting the echo to stderr?

Reply:
Indenting the body of the the remediation didn't break the build.
Unfortunately, not everything is indented.
When the scanner performs the XCCDF substitutions, the Bash remediation functions are not properly indented.
Main issue being that they have multiple lines, and can be used "platform wrapped" or not.

Reply:
@matejak `indent` is not available in python2.
Do you know of a trivial substitute?

Reply:
Oh, that Python2 not being able to do this - that's a surprise. We can have an indent as an optional feature in the try-except block, but I wouldn't do it now, as it conflicts with XCCDF substitutions, and there is a small likelihood that some remediations may not be indent-safe - consider [here-docs](https://unix.stackexchange.com/questions/76481/cant-indent-heredoc-to-match-code-blocks-indentation).

Reply:
Interesting find. I'll drop adcb827 for now.